src/definitions/bluebps.json

```diff
@@ -0,0 +1,276 @@
+                        "units": "millisecond"
+                    }
+                ]
+            },
+            "temperature_max": {
+                "id": "9104",
+                "description": "Get the maximum allowed battery temperature",
+                "payload": [
+                    {
+                        "name": "limit",
+                        "type": "u16",
+                        "description": "The minimum voltage allowed for any individual cell. 0~5000: 0~5V",
+                        "units": "millivolt"
+                    }
+                ]
+            },
+            "temperature_timeout": {
+                "id": "9105",
+                "description": "Get the over-temperature timeout",
+                "payload": [
+                    {
+                        "name": "timeout",
+                        "type": "u16",
+                        "description": "If the battery temperature exceeds the configured limit for this duration of time, the power will be locked-out",
+                        "units": "millisecond"
+                    }
+                ]
+            },
+            "state": {
+                "id": "9106",
+                "description": "Get the current state of the device",
+                "payload": [
+                    {
+                        "name": "battery_voltage",
+                        "type": "u16",
+                        "description": "The main battery voltage",
+                        "units": "millivolt"
+                    },
+                    {
+                        "name": "battery_current",
+                        "type": "u16",
+                        "description": "The current measurement",
+                        "units": "centiampere"
+                    },
+                    {
+                        "name": "battery_temperature",
+                        "type": "u16",
+                        "description": "The battery temperature",
+                        "units": "decidegrees Centigrade"
+                    },
+                    {
+                        "name": "cpu_temperature",
+                        "type": "u16",
+                        "description": "The cpu temperature",
+                        "units": "decidegrees Centigrade"
+                    },
+                    {
+                        "name": "flags",
+                        "type": "u8",
+                        "description": "flags indicating if any of the configured limits are currently exceeded"
+                    },
+                    {
+                        "name": "cell_voltages",
+                        "type": "vector",
+                        "vector": {
+                            "sizetype": "u8",
+                            "datatype": "u16",
+                            "size": "dynamic"
```

Comment:
How is this dynamic ? Where is it possible to configure it ? Is it an automatic thing ? If so how ?

Reply:
cell count is determined at runtime. The r2 hardware supports 3s - 6s. Future hardwares may support more.

Reply:
The number of cells is detected automatically with adc channels on balance plug.

Reply:
My only concern is if this is safe with a broken cell connection between battery - BPS, resulting in wrong automatic cell detection

Reply:
Yes we will need to figure out some logic on how to detect and handle errors.